### PR TITLE
tilt: p is for profile

### PR DESCRIPTION
**Pull request by landism**
'p' now starts and stops a pprof profile

### CODE REVIEW

File: internal/store/engine_state.go [hunk 1]
```diff
@@ -65,6 +65,9 @@ type EngineState struct {
 
 	TriggerMode  model.TriggerMode
 	TriggerQueue []model.ManifestName
+
+	Profiling   bool
+	ProfileFile *os.File
```
Comment (nicks) [inline internal/store/engine_state.go]: it seems like a bad idea to have a pointer to an external OS thing in the engineState, especially as we start cloning/serializing/copying this state to the web ui, protobuf, etc

File: internal/hud/hud.go [hunk 2]
```diff
@@ -37,6 +38,8 @@ type Hud struct {
 	currentViewState view.ViewState
 	mu               sync.RWMutex
 	isRunning        bool
+
+	currentProfileFile *os.File
```
Comment (nicks) [inline internal/hud/hud.go]: is this used anywhere?

**Comment by maiamcc** [comment]
https://www.youtube.com/watch?v=_dvPhtNZCj0

File: internal/store/engine_state.go (hunk 1 above)
Comment (landism) [reply internal/store/engine_state.go]: Oh, I didn't realize that serializability was a requirement for EngineState, and I didn't really see anywhere else Upper stores state. I'll inject something into Upper to hold the state.

File: internal/hud/hud.go (hunk 2 above)
Comment (landism) [reply internal/hud/hud.go]: cruft from an earlier iteration. thanks!

**Pull request merged**